### DISCUSSION TITLE
[HELP]/[BUGS/ERRORS] getting started hello prism fails

### DISCUSSION THREAD

**Discussion by OlagStegan** [prisma/prisma#6881] [Q&A] [answered]
OK I am new to Prisma so I went to the docs and "getting started.
On a Windows 10 PC
Nodejs v16.0.0 installed and MySQL and MariaDB @ latest versions
Follow the instructions all the way to running the node index.js everything works perfect.
I can confirm that the tables are created in the new database by using my preferred client (Navicat).
But when I run node index.js I get the following error output:

`PS D:\WEB\_nodejs> cd hello-prisma
PS D:\WEB\_nodejs\hello-prisma> npx prisma migrate dev --name init
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": MySQL database "hello" at "localhost:3306"

MySQL database hello created at localhost:3306

The following migration(s) have been created and applied from new schema changes:

migrations/
  └─ 20210501084754_init/
    └─ migration.sql

Your database is now in sync with your schema.

Running generate... (Use --skip-generate to skip the generators) 

up to date, audited 3 packages in 2s

found 0 vulnerabilities

added 2 packages, and audited 5 packages in 5s

✔ Installed the @prisma/client and prisma packages in your projec✔ Generated Prisma Client (2.21.2) to .\node_modules\@prisma\client in 316ms

PS D:\WEB\_nodejs\hello-prisma> npm install @prisma/client       

up to date, audited 5 packages in 2s

found 0 vulnerabilities
PS D:\WEB\_nodejs\hello-prisma> node index.js
C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe[2248]: 
c:\ws\src\node_http_parser.cc:567: Assertion `args[3]->IsInt32()' failed.
 1: 00007FF6180686BF v8::internal::Isolate::ArchiveSpacePerThread+4719
 2: 00007FF617FFB246 v8::internal::wasm::WasmCode::safepoint_table_offset+67414
 3: 00007FF617FFB5C1 v8::internal::wasm::WasmCode::safepoint_table_offset+68305
 4: 00007FF617FE0543 v8::base::CPU::has_sse+14915
 5: 00007FF618923969 v8::internal::Builtins::builtin_handle+318073
 6: 00007FF618922F01 v8::internal::Builtins::builtin_handle+315409
 7: 00007FF6189231F1 v8::internal::Builtins::builtin_handle+316161
 8: 00007FF618923033 v8::internal::Builtins::builtin_handle+315715
 9: 00007FF6189F98B1 v8::internal::SetupIsolateDelegate::SetupHeap+455217
10: 00007FF6189939B9 v8::internal::SetupIsolateDelegate::SetupHeap+37689
11: 00007FF618990DFF v8::internal::SetupIsolateDelegate::SetupHeap+26495
12: 00007FF618A853DA v8::internal::SetupIsolateDelegate::SetupHeap+1027418
13: 00007FF6189939B9 v8::internal::SetupIsolateDelegate::SetupHeap+37689
14: 00007FF6189939B9 v8::internal::SetupIsolateDelegate::SetupHeap+37689
15: 00007FF6189939B9 v8::internal::SetupIsolateDelegate::SetupHeap+37689
16: 00007FF6189939B9 v8::internal::SetupIsolateDelegate::SetupHeap+37689
17: 00007FF6189939B9 v8::internal::SetupIsolateDelegate::SetupHeap+37689
18: 00007FF6189939B9 v8::internal::SetupIsolateDelegate::SetupHeap+37689
19: 00007FF6189939B9 v8::internal::SetupIsolateDelegate::SetupHeap+37689
20: 00007FF618A3FA48 v8::internal::SetupIsolateDelegate::SetupHeap+742344
21: 00007FF618990FCC v8::internal::SetupIsolateDelegate::SetupHeap+26956
22: 00007FF618A853DA v8::internal::SetupIsolateDelegate::SetupHeap+1027418
23: 00007FF6189939B9 v8::internal::SetupIsolateDelegate::SetupHeap+37689
24: 00007FF6189939B9 v8::internal::SetupIsolateDelegate::SetupHeap+37689
25: 00007FF6189BFCB3 v8::internal::SetupIsolateDelegate::SetupHeap+218675
26: 00007FF618A413A5 v8::internal::SetupIsolateDelegate::SetupHeap+748837
27: 00007FF6189B3513 v8::internal::SetupIsolateDelegate::SetupHeap+167571
28: 00007FF618991B0B v8::internal::SetupIsolateDelegate::SetupHeap+29835
29: 00007FF61884A0E3 v8::internal::Execution::CallWasm+1891      
30: 00007FF61884A1EB v8::internal::Execution::CallWasm+2155      
31: 00007FF61884AC5A v8::internal::Execution::TryCall+378        
32: 00007FF618827945 v8::internal::MicrotaskQueue::RunMicrotasks+501
33: 00007FF6188276A0 v8::internal::MicrotaskQueue::PerformCheckpoint+32
34: 00007FF618923969 v8::internal::Builtins::builtin_handle+318073
35: 00007FF618922F01 v8::internal::Builtins::builtin_handle+315409
36: 00007FF6189231F1 v8::internal::Builtins::builtin_handle+316161
37: 00007FF618923033 v8::internal::Builtins::builtin_handle+315715
38: 00007FF6189F98B1 v8::internal::SetupIsolateDelegate::SetupHeap+455217
39: 00007FF6189939B9 v8::internal::SetupIsolateDelegate::SetupHeap+37689
40: 00007FF618991C0F v8::internal::SetupIsolateDelegate::SetupHeap+30095
41: 00007FF61899180B v8::internal::SetupIsolateDelegate::SetupHeap+29067
42: 00007FF61884A022 v8::internal::Execution::CallWasm+1698
43: 00007FF61884985F v8::internal::Execution::Call+191
44: 00007FF618950EF1 v8::Function::Call+609
45: 00007FF618097878 node::CallbackScope::~CallbackScope+888     
46: 00007FF618097BEB node::CallbackScope::~CallbackScope+1771
47: 00007FF61808F714 v8::internal::compiler::Operator::EffectOutputCount+228
48: 00007FF617F19A53 RSA_meth_get_flags+20675
49: 00007FF617F1A5CD RSA_meth_get_flags+23613
50: 00007FF617F14EF9 RSA_meth_get_flags+1385
51: 00007FF6180B9E62 uv_thread_self+3986
52: 00007FF6180BCBEF uv_pipe_pending_type+3359
53: 00007FF6180C798C uv_loop_init+940
54: 00007FF6180C7C8A uv_run+202
55: 00007FF618096F64 node::SpinEventLoop+308
56: 00007FF617FB3F53 v8::internal::UnoptimizedCompilationInfo::feedback_vector_spec+52995
57: 00007FF61802E387 node::Start+215
58: 00007FF617E5822C RC4_options+346412
59: 00007FF618F1931C v8::internal::compiler::RepresentationChanger::Uint32OverflowOperatorFor+151692
60: 00007FFFC5C57034 BaseThreadInitThunk+20
61: 00007FFFC7AC2651 RtlUserThreadStart+33.`

I do not understand if this is really a bug or something not covered or missing in the docs.

This does not seem to have been reported before.
I have now repeated the process in a new clean dir and it is reproducible with same errors. I am hoping someone here can make sense of this and help advise next steps before giving up on Prisma all together.

Thanks in advance.

**Comment by pr0ph0z** [comment]
Not really sure but as I read it's about Node.js system core. How about trying to downgrading your Node.js to an LTS version?

**Reply by OlagStegan** [reply]
You might well be right but downgrading Node is not really an option as the version is required for other projects. I was following the "system requirements" and they do not specify the Node version as LTS.

the error that caught my eye was the "Assertion args[3]->IsInt32()' failed." which looks like a field type/definition

**Comment by shirshak55** [comment, marked as answer]
Downgrade node or upgrade prisma to dev

I solved it by using latest dev version

yarn add @prisma/client@dev prisma@dev

For me it is:
```
  "@prisma/client": "^2.22.0-dev.48",
  ```

**Reply by OlagStegan** [reply]
Many thanks that has cured the problem. I can now move on to better things.
Made my weekend!

**Reply by pantharshit00** [reply]
For the record, this is caused by node 16 upgrade and we fixed it already via #6682. This change will next in the stable release next Tuesday.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
